modules/artifact/src/test/java/org/mule/runtime/module/artifact/classloader/IBMMQResourceReleaserTestCase.java

```diff
@@ -210,6 +200,201 @@ public void mBeansTest() throws Exception {
+    assertThat(countWorkerThreads(artifactClassLoaderApplication1), is(0));
+    assertThat(countWorkerThreads(artifactClassLoaderApplication2), is(1));
+    assertThat(countWorkerThreads(artifactClassLoaderApplication3), is(1));
+
+    artifactClassLoaderApplication2.dispose();
+
+    assertThat(countWorkerThreads(muleDomainClassLoader), is(0));
+    assertThat(countWorkerThreads(artifactClassLoaderApplication1), is(0));
+    assertThat(countWorkerThreads(artifactClassLoaderApplication2), is(0));
+    assertThat(countWorkerThreads(artifactClassLoaderApplication3), is(1));
+
+    artifactClassLoaderApplication3.dispose();
+
+    assertThat(countWorkerThreads(muleDomainClassLoader), is(0));
+    assertThat(countWorkerThreads(artifactClassLoaderApplication1), is(0));
+    assertThat(countWorkerThreads(artifactClassLoaderApplication2), is(0));
+    assertThat(countWorkerThreads(artifactClassLoaderApplication3), is(0));
+
+  }
+
+  @Test
+  @Description("When removing an application which contains the IBM MQ Driver, there should not be worker thread references left")
+  public void threadWorkerDomainDriver() throws Exception {
+    MuleArtifactClassLoader muleDomainClassLoader =
+        new MuleArtifactClassLoader("IBMMQResourceReleaserTestCase1",
+                                    mock(ArtifactDescriptor.class),
+                                    new URL[] {getBundleDependency().getBundleUri().toURL()},
+                                    currentThread().getContextClassLoader(),
+                                    testLookupPolicy);
+
+    // Creating Worker Thread domain level
+    createWorkerThread(muleDomainClassLoader);
+
+    MuleArtifactClassLoader artifactClassLoaderApplication1 =
+        new MuleArtifactClassLoader("IBMMQResourceReleaserTestCase2",
+                                    mock(ArtifactDescriptor.class),
+                                    new URL[] {getBundleDependency().getBundleUri().toURL()},
+                                    muleDomainClassLoader,
+                                    testLookupPolicy);
+
+    MuleArtifactClassLoader artifactClassLoaderApplication2 =
+        new MuleArtifactClassLoader("IBMMQResourceReleaserTestCase3",
+                                    mock(ArtifactDescriptor.class),
+                                    new URL[] {getBundleDependency().getBundleUri().toURL()},
+                                    muleDomainClassLoader,
+                                    testLookupPolicy);
+
+    MuleArtifactClassLoader artifactClassLoaderApplication3 =
+        new MuleArtifactClassLoader("IBMMQResourceReleaserTestCase4",
+                                    mock(ArtifactDescriptor.class),
+                                    new URL[] {getBundleDependency().getBundleUri().toURL()},
+                                    muleDomainClassLoader,
+                                    testLookupPolicy);
+
+    assertThat(countWorkerThreads(muleDomainClassLoader), is(1));
```

Comment:
I would do a disposal of one of the app classloaders to showcase that the thread of the domain is still alive.

Reply:
This is done look line 325

Reply:
You are right, my bad